upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go

```diff
@@ -128,7 +128,15 @@ func (t *LaunchTemplate) RenderAWS(c *awsup.AWSAPITarget, a, ep, changes *Launch
 		}
 		lc.UserData = aws.String(base64.StdEncoding.EncodeToString(d))
 	}
-
+	// @step: add instanceInterruptionBehavior
+	if t.InstanceInterruptionBehavior != nil {
+		s := &ec2.LaunchTemplateSpotMarketOptionsRequest{
+			InstanceInterruptionBehavior: t.InstanceInterruptionBehavior,
+		}
+		lc.InstanceMarketOptions = &ec2.LaunchTemplateInstanceMarketOptionsRequest{
+			SpotOptions: s,
+		}
+	}
```

Comment:
this looks correct to me but I haven't tested it. Is it giving you errors?

Reply:
@rifelpet I have commited all the code suggestions and now the tests passed. So for me it looks good.